pkg/exchange/okex/exchange.go

```diff
@@ -363,3 +378,116 @@ func (e *Exchange) QueryOrder(ctx context.Context, q types.OrderQuery) (*types.O
 }
+
+/*
+Query Closed trades can query closed trades in last 3 months, there are no time interval limitations, as long as until >= since.
+Please Use LastTradeID as cursor, only return trades earlier than that trade, that trade is not included.
+If you want to query trades by time range, please just pass since and until.
+If you want to query by cursor, please pass LatestTradeID.
+Because it get correct response even pass all parameters with right time interval and invalid trade id, like 0.
+*/
+func (e *Exchange) QueryClosedOrders(ctx context.Context, symbol string, since, until time.Time, lastOrderID uint64) ([]types.Order, error) {
+	if symbol == "" {
+		return nil, ErrSymbolRequired
+	}
+
+	if err := tradeRateLimiter.Wait(ctx); err != nil {
+		return nil, fmt.Errorf("query closed order rate limiter wait error: %w", err)
+	}
+
+	var lastOrder string = strconv.FormatUint(lastOrderID, 10)
+
+	res, err := e.client.NewGetOrderHistoryRequest().
+		InstrumentID(toLocalSymbol(symbol)).
+		StartTime(since).
+		EndTime(until).
+		Limit(defaultQueryLimit).
+		After(lastOrder).
+		Do(ctx)
+
+	if err != nil {
+		return nil, fmt.Errorf("failed to call get order histories error: %w", err)
+	}
+
+	var orders []types.Order
+
+	for _, order := range res {
+		o, err2 := toGlobalOrder(&order)
+		if err2 != nil {
+			err = multierr.Append(err, err2)
+			continue
+		}
+
+		if o.Status.Closed() {
+			orders = append(orders, *o)
+		}
+	}
+	if err != nil {
+		return nil, err
+	}
+
+	return types.SortOrdersAscending(orders), nil
+}
+
+/*
+OKEX do not provide api to query by tradeID, So use /api/v5/trade/orders-history-archive as its official site do.
+Query trades can query trades in last 3 months, there are no time interval limitations, as long as end_time >= start_time.
+Please Use LastTradeID as cursor, only return trades earlier than that trade, that trade is not included.
+If you want to query trades by time range, please just pass start_time and end_time.
+If you want to query by cursor, please pass LatestTradeID.
+Because it get correct response even we pass all parameters with right time interval and invalid trade id, like 0.
+*/
+func (e *Exchange) QueryTrades(ctx context.Context, symbol string, options *types.TradeQueryOptions) ([]types.Trade, error) {
+	if symbol == "" {
+		return nil, ErrSymbolRequired
+	}
+
+	req := e.client.NewGetOrderHistoryRequest().InstrumentID(toLocalSymbol(symbol))
+
+	limit := uint64(options.Limit)
+	if limit > defaultQueryLimit || limit <= 0 {
+		log.Debugf("limtit is exceeded or zero, update to %d, got: %d", defaultQueryLimit, options.Limit)
+		limit = defaultQueryLimit
+	}
+	req.Limit(limit)
+
+	if err := tradeRateLimiter.Wait(ctx); err != nil {
+		return nil, fmt.Errorf("query trades rate limiter wait error: %w", err)
+	}
+
+	var lastTradeID string = strconv.FormatUint(options.LastTradeID, 10)
+
+	var response []okexapi.OrderDetails
+	// query by time interval
+	if options.StartTime != nil || options.EndTime != nil {
```

Comment:
what if both options.StartTime == nil and options.EndTime == nil here

Reply:
Update ! Need to consider this situation, and I add its unit test ~